ci-infra/azure/variables.tf

```diff
@@ -0,0 +1,55 @@
+variable "ver" {
```

Comment:
What is the reasoning for this variable?

Reply:
I added `ver` so that it becomes easier to switch, if we needed to. From one set up to a new one. The problem with having consistent naming shows up in recreating storage accounts, container registries, etc.

Reply:
Is there a better way to tackle these problems?

Reply:
I'm not sure I understood the problem yet, you intend to keep multiple versions of a terraform deployment? If that's the case, wouldn't bumping the ver variable just recreate the resources with new names?

Reply:
and if there are name clashes with non-tf managed resources does `random_id()` help?

Reply:
> you intend to keep multiple versions of a terraform deployment?

Yes for a short while before tearing down old set up and still want to recreate a new set up.

Reply:
So that's not what you want, right? A `tf apply` would remove the old resources and create the new ones in one go. If I understand correctly you want to have some kind of blue/green deployment?

Reply:
Yes, you are right. Hmm, makes me wonder probably we don't need this everywhere. Only for things that will conflict, like storage account, registry name, etc.

Reply:
I think even with clean up it is fine. As in we will have a whole new set up every time we run it. Especially in environments where things will require no sharing with previous setups.